python/python/bystro/prs/tests/test_preprocess_for_prs.py

```diff
@@ -88,48 +94,52 @@ def mock_bin_mappings():
-        {
-            "chr10:10082621:G:T": [0, 2],
-            "chr10:10528970:A:G": [1, 0],
-        },
-        index=["Sample1", "Sample2"],
-    )
+def mock_finalize_scores_after_c_t():
+    def _mock_finalize_scores_after_c_t(
+        gwas_scores_path, dosage_matrix_path, map_directory_path, p_value_threshold  # noqa: ARG001
+    ):
+        scores_after_c_t = pd.DataFrame(
+            {"BETA": [0.007630, -0.020671], "P": [0.699009, 0.0030673]},
+            index=["chr1:566875:C:T", "chr1:728951:A:G"],
+        )
+        loci_and_allele_comparison = {"chr1:566875:C:T": ("C", "T"), "chr1:728951:A:G": ("G", "A")}
+        return scores_after_c_t, loci_and_allele_comparison
 
-    scores_overlap_adjusted_mock = pd.DataFrame(
-        {
-            "SNPID": ["chr10:10082621:G:T", "chr10:10528970:A:G"],
-            "BETA": [0.1, -0.2],
-        }
-    ).set_index("SNPID")
+    return _mock_finalize_scores_after_c_t
 
-    expected_results = pd.Series(
-        {
-            "Sample1": (0 * 0.1) + (1 * -0.2),
-            "Sample2": (2 * 0.1) + (0 * -0.2),
+
+@pytest.fixture()
+def mock_read_feather_in_chunks():
+    def _mock_read_feather_in_chunks(dosage_matrix_path, columns=None, chunk_size=1000):  # noqa: ARG001
+        data = {
+            "locus": ["chr1:566875:C:T", "chr1:728951:A:G", "chr1:917492:C:T", "chr2:917492:A:T"],
+            "ID00096": [1, 1, 2, -1],
+            "ID00097": [0, 1, 1, 0],
         }
-    )
-    return genos_transpose_mock, scores_overlap_adjusted_mock, expected_results
-
-
-def test_extract_nomiss_dosage_loci(mock_dosage_df):
-    filtered_df = mock_dosage_df[mock_dosage_df["ID00096"] != -1]
-    mock_table = pa.Table.from_pandas(filtered_df)
-    with patch("pyarrow.dataset.dataset") as mock_dataset:
-        mock_ds = MagicMock()
-        mock_ds.schema.names = mock_dosage_df.columns.tolist()
-        mock_ds.filter.return_value = mock_ds
-        mock_ds.to_table.return_value = mock_table
-        mock_dataset.return_value = mock_ds
-        result = _extract_nomiss_dosage_loci("mock/path/to/dosage.feather")
-
-        assert not result.empty, "The DataFrame should not be empty."
-        assert "locus" in result.columns, "'locus' column should be present in the DataFrame."
-        assert (
-            len(result) == 3
-        ), "The number of rows in the DataFrame should be 3."
-        mock_dataset.assert_called_once_with("mock/path/to/dosage.feather", format="feather")
+        chunk = pd.DataFrame(data)
+        yield chunk
+
+    return _mock_read_feather_in_chunks
+
+
+@pytest.fixture()
+def mock_finalize_dosage_after_c_t():
+    def _mock_finalize_dosage_after_c_t(chunk, loci_and_allele_comparison):  # noqa: ARG001
+        return pd.DataFrame(
+            {"ID00096": [1, 1], "ID00097": [0, 1]}, index=["chr1:566875:C:T", "chr1:728951:A:G"]
+        ).transpose()
+
+    return _mock_finalize_dosage_after_c_t
+
+
+def test_extract_nomiss_dosage_loci(mock_dosage_df, mock_scores_loci):
```

Comment:
```suggestion
def test_extract_nomiss_dosage_loci(tmp_path):
    # Create a temporary Feather file for testing
    test_file = tmp_path / "test_dosage_matrix.feather"
    data = {
        'locus': ['locus1', 'locus2', 'locus3', 'locus4'],
        'sample1': [0.1, -1, 0.3, 0.4],
        'sample2': [0.2, 0.3, -1, 0.5],
        'sample3': [0.3, 0.4, 0.5, 0.6]
    }
    table = pa.table(data)
    pa.feather.write_feather(table, test_file)
    
    score_loci = {'locus1', 'locus3', 'locus4'}
    
    result_loci = _extract_nomiss_dosage_loci(test_file, score_loci)
    # Only locus4 has no missing values
    assert result_loci == {'locus1', 'locus4'}
```

I prefer testing end-end where possible; in the suggested test we read from disk, as we would in the actual use

Reply:
Ok, should I use a temp feather file everywhere that it goes into the actual functions? I was trying to avoid I/O for tests in general so that they run quickly, which is why I was trying to avoid using temp feather files, though if you think that's better I can start including I/O in tests.